metrics/benchmark_test.go

```diff
@@ -10,19 +10,39 @@ import (
 	"testing"
 )
 
+func BenchmarkNewTag(b *testing.B) {
+	b.Run("1 tag", func(b *testing.B) {
+		doTagBench(b, 1)
+	})
+	b.Run("10 tag", func(b *testing.B) {
+		doTagBench(b, 10)
+	})
+	b.Run("100 tag", func(b *testing.B) {
+		doTagBench(b, 100)
+	})
+}
+
+func doTagBench(b *testing.B, n int) {
+	var tags Tags
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		tags = append(tags, MustNewTag(fmt.Sprintf("key%d", i), fmt.Sprintf("val%d", i)))
```

Comment:
A lot of the time in the benchmark is going into growing this array. Just assigning the result to `_` will run the function with less happening that you aren't trying to observe
```suggestion
		n := strconv.Itoa(i)
		_ = MustNewTag("key"+n, "val"+n)
```